src/components/flash/flash.js

```diff
@@ -133,9 +135,16 @@ class Flash extends React.Component {
    * @method componentWillReceiveProps
    * @return(Void)
    */
-  componentWillReceiveProps(prevProps) {
-    if (prevProps.open !== this.props.open) {
-      this.setState({ dialogs: {} });
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.open === this.props.open) { return; }
+
+    if (nextProps.open) {
+      this.setState({ dialogs: {}, open: nextProps.open });
```

Comment:
I'm curious why we need `this.state.open`, when all we seem to do is set it to the value of `this.props.open`?

Can we not just use `this.props.open`?

Reply:
If we did this ```this.prop.open && (
        <Portal>``` Portal and content are automatically removed. ```clearTimeout(this.fadeLeaveTimeout);
      this.fadeLeaveTimeout = setTimeout(() => {
        this.setState({ open: nextProps.open });
      }, 2000);``` gives us latency for the fadeout to operate before the Portal and children are torn down. If you can see a better way, sincerely I am all ears.